pkg/ingester/ingester_v2.go

```diff
@@ -1057,8 +1064,13 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
 	wg := &sync.WaitGroup{}
 	openGate := gate.New(i.cfg.BlocksStorageConfig.TSDB.MaxTSDBOpeningConcurrencyOnStartup)
 
-	err := filepath.Walk(i.cfg.BlocksStorageConfig.TSDB.Dir, func(path string, info os.FileInfo, err error) error {
+	// Keep track of all errors that could occur.
+	errs := tsdb_errors.MultiError{}
+	errsMx := sync.Mutex{}
+
+	walkErr := filepath.Walk(i.cfg.BlocksStorageConfig.TSDB.Dir, func(path string, info os.FileInfo, err error) error {
 		if err != nil {
+			level.Warn(util.Logger).Log("msg", "skipped filesystem entry when looking for existing TSDB to open", "path", path, "err", err)
 			return filepath.SkipDir
```

Comment:
If there is any error traversing the tree, shouldn't we return such error?  (Esp. if we're halting ingester if we fail to open TSDB)

Reply:
Yes, better to do error out. Done.